insonmnia/hardware/hardware.go

```diff
@@ -68,6 +80,94 @@ func (h *Hardware) SetNetworkIncoming(IPs []string) {
 	}
 }
 
+func (h *Hardware) AskPlanResources() *sonm.AskPlanResources {
+	result := sonm.NewEmptyAskPlanResources()
+	result.CPU.CorePercents = uint64(h.CPU.GetDevice().GetCores()) * 100
+	result.RAM.Size.Bytes = h.RAM.Device.Available
+	result.Storage.Size.Bytes = h.Storage.GetDevice().GetBytesAvailable()
+	for _, gpu := range h.GPU {
+		result.GPU.Hashes = append(result.GPU.Hashes, gpu.Device.Hash)
+	}
+	result.Network.Outbound = h.Network.Outbound
+	result.Network.Overlay = h.Network.Overlay
+	result.Network.Incoming = h.Network.Incoming
+	//TODO: Make network device use DataSizeRate
+	result.Network.ThroughputIn.BitsPerSecond = h.Network.GetIn()
+	result.Network.ThroughputOut.BitsPerSecond = h.Network.GetOut()
+	return result
+}
+
+func insertBench(to []uint64, bench *sonm.Benchmark, proportion float64) ([]uint64, error) {
+	for len(to) <= int(bench.ID) {
+		to = append(to, uint64(0))
```

Comment:
Not sure, but may be this is more clear:
```go
to = append(to, make([]uint64, len(to) - int(bench.ID))...)
```

Reply:
you have a bug in your example)

Reply:
My bad

Reply:
any better ideas?)